sklearn/preprocessing/data.py

```diff
@@ -1797,6 +1801,20 @@ def _fit_transform(self, X):
             out = out[:, active_features]
             self.active_features_ = active_features
 
+            # Find the index of the start of each class in the
+            # `active_features` array
+            split_indices = np.searchsorted(active_features,
+                                            self.feature_indices_)
+
+            af_copy = active_features.copy()
```

Comment:
Because after the call to `np.split` the individual slices are modified and we don't want it to affect the original array.